index.js

```diff
@@ -0,0 +1,179 @@
+const private_repositories = core.getInput('private_repositories');
+const branches = core.getInput('branches');
+const theme_repository  = core.getInput('theme_repository');
+const theme_branch  = core.getInput('theme_branch');
+
+async function run() {
+  try {
+
+      // Create and activate virtualenv
+      await exec.exec('python3', ['-m', 'venv', 'venv'], options);
+      core.info('Virtualenv created');
+
+      // Install Tutor
+      core.info('Installing Tutor');
+      await exec.exec('venv/bin/python', ['-m', 'pip', 'install', `tutor==${tutor_version}`], options);
+
+      // Install the Tutor Pearson Plugin
+      if (tutor_pearson_plugin_url) {
+          core.info('Installing Tutor Pearson plugin');
+          const gh_token_url = tutor_pearson_plugin_url.replace("GH_TOKEN", gh_access_token);
+          await exec.exec('venv/bin/pip', ['install', gh_token_url], options);
+      }
+
+      // Install Tutor plugins
+      if (tutor_plugin_sources) {
+          core.info('Installing Tutor plugins');
+          const plugin_sources = parse_bash_array(tutor_plugin_sources);
+          for (var i=0; i < plugin_sources.length; i++) {
+              let plugin_source = plugin_sources[i];
+              await exec.exec('venv/bin/pip', ['install', plugin_source], options);
+          }
+      }
+
+      // Path to the tutor root will be saved in this variable
+      var tutor_root = "";
+      var tutor_root_options = {
+        shell: '/bin/bash'
+      };
+      tutor_root_options.listeners = {
+        stdout: (data) => {
+          tutor_root += data.toString();
+        },
+        stderr: (data) => {
+          myError += data.toString();
+        }
+      };
+
+      await exec.exec('venv/bin/tutor', ['config', 'save'], options);
+
+      // Running 'tutor config printroot' and saving the output to a variable
+      await exec.exec('venv/bin/tutor', ['config', 'printroot'], tutor_root_options);
+      tutor_root = tutor_root.trim();
+      core.info(`tutor_root: ${tutor_root}`);
+
+      // Create private.txt file
+      await exec.exec('touch', [`${tutor_root}/env/build/openedx/requirements/private.txt`], options);
+
+      // Install extra requirements
+      if (extra_private_requirements) {
+        core.info('Installing extra private requirements');
+        const repositories = parse_bash_array(private_repositories);
+        const branches_array = parse_bash_array(branches);
+        for (var i=0; i< repositories.length; i++) {
+          let repository = repositories[i];
+          let branch = branches_array[i];
+          if (repository == "") {
+            continue;
+          }
+          await exec.exec(
+            'git',
+            [
+              'clone', '-b', branch,
+              `https://${gh_access_token}@github.com/Pearson-Advance/${repository}.git`,
+              `${tutor_root}/env/build/openedx/requirements/${repository}`
+            ],
+            options
+          );
```

Comment:
What happens is repo is not "" but also not a really repository, does it stop the whole procedure?. If branch is "", what should we do?. @ShonTitor 
```suggestion
        const branches = parse_bash_array(branches);
        for (var i=0; i< repositories.length; i++) {
          let repository = repositories[i];
          if (repository == "") {
            continue;
          }
          await exec.exec(
            'git',
            [
              'clone', '-b', branches[i],
              `https://${gh_access_token}@github.com/Pearson-Advance/${repository}.git`,
              `${tutor_root}/env/build/openedx/requirements/${repository}`
            ],
            options
          );
```

Reply:
It is true that problems can arise if the lengths of the arrays don't match. For these cases, I think the best approach would be to check if the branch name is truthy and to run the command without the `-b` flag if it isn't.